src/main/java/frc/robot/systems/DriveFSMSystem.java

```diff
@@ -0,0 +1,233 @@
+	 * when the robot is enabled.
+	 *
+	 * Note this is distinct from the one-time initialization in the constructor
+	 * as it may be called multiple times in a boot cycle,
+	 * Ex. if the robot is enabled, disabled, then reenabled.
+	 */
+	public void reset() {
+		currentState = FSMState.TELEOP_STATE;
+
+		// Call one tick of update to ensure outputs reflect start state
+		update(null);
+	}
+	/**
+	 * Update FSM based on new inputs. This function only calls the FSM state
+	 * specific handlers.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 */
+	public void update(TeleopInput input) {
+		switch (currentState) {
+			case START_STATE:
+				handleStartState(input);
+				break;
+
+			case TELEOP_STATE:
+				handleTeleOpState(input);
+                break;
+                
+			case FORWARD_STATE_10_IN:
+				handleForwardOrBackwardState(input, 10);
+				break;
+
+			case TURN_STATE:
+				handleTurnState(input, 90); // test 90 degrees
+				break;
+
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+		currentState = nextState(input);
+	}
+
+	/* ======================== Private methods ======================== */
+	/**
+	 * Decide the next state to transition to. This is a function of the inputs
+	 * and the current state of this FSM. This method should not have any side
+	 * effects on outputs. In other words, this method should only read or get
+	 * values to decide what state to go to.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 * @return FSM state for the next iteration
+	 */
+	private FSMState nextState(TeleopInput input) {
+		switch (currentState) {
+			case START_STATE:
+				if (input != null) {
+					return FSMState.TELEOP_STATE;
+				} else {
+					return FSMState.START_STATE;
+				}
+
+			case TELEOP_STATE:
+				return FSMState.TELEOP_STATE;
+            
+			case FORWARD_STATE_10_IN:
+				return FSMState.TURN_STATE;
+				//should be return FSMState.TURN_RIGHT; once simon pushes his code
+
+			case TURN_STATE:
+				return FSMState.START_STATE;
+
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+	}
+
+	/* ------------------------ FSM state handlers ------------------------ */
+	/**
+	 * Handle behavior in START_STATE.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 */
+	private void handleStartState(TeleopInput input) {
+		setPowerForAllMotors(0);
+	}
+
+	//Assume encoder starts at 0.
+	/**
+	* Handle behavior in FORWARD_STATE, or BACKWARD_STATE
+	* @param input Global TeleopInput if robot in teleop mode or null if
+	*        the robot is in autonomous mode.
+	* @param inches The number of inches to move forward or backward
+	*/
+
+	//Assume encoder positions are at 0 initially
```

Comment:
You can make sure this is true, but resetting the encoders in all the states where they aren't used.